Comic scraper

RascalTwo opened the pull request:
A partly-modular comic scraping cog.

Currently allows for each comic website that wishes to be scraped by adding a new file.

*****

I personally see it as quite newbie friendly, and have added a second example comic to be scraped when it was suggested that indirection was present.

A alternative to this I had before, was the structure in commit https://github.com/PythonBuddies/main_bot/commit/b8caa632dbb91794ff1315c97a55e0c67ad984f7, in which it's all within a single file.

I passed up this method due to the fact that this would make the file 100s of lines long, and - since this is due to be the newbie-friendly task - it is easier to read at the current point.

*****

No matter what structure is chosen, `comic random` will pick a random comic website, and execute the `random` command within it, returning a random comic.

`comic latest` will also cycle through all loaded comic websites, executing the `latest` command for each of them.

*****

I have also tried my hand at writing up documentation for this, along with a code-walkthrough in which I attempt to explain everything the code is doing.

There is also a - although currently raw - how-to for scraping a website under `ComicScraper_Scraping.md`

*****

I believe leaving this pull request open until Friday or Monday will be a reasonable amount of time for counter-points and suggestions to be brought up.

I plan to squash a handful of these commits, format/rewrite the `ComicScraper_Scraping.md` file, and - maybe - add documentation files for both the `commitstrip.py` and `cubedrone.py` files, if need be, before the pull request is merged.

*****

Well this has been the longest pull request I've written, so thanks for reading! :smiley:

raylu commented:
"modular with inheritance from a base scraper" vs "all comic scrapers in one file" seems like a false choice. why not keep the scrapers separate and keep the classes separate too? why does there need to be _indirection_ and especially why is there reflection?

RascalTwo commented:
I think I fixed what you're pointing out.

*****

Made the comic scrapers themselves access the `ComicScraper` cog and add their `_latest` and `_random` commands manually, so no more searching the `comic` command for `_random` or `_latest` commands every time.

Also changed the `dir` structure, so the main `comic_scraper.py` is outside the `comics` dir, and only the scrapers themselves are in the `comics` dir.

*****

I could go as far as removing the `ComicScraper` import from the comic modules:

```Python
from ..comic_scraper import ComicScraper
```

Although I don't think it'd be possible to use descriptors if I did so, so one would have to manually create all the commands in the `setup` method.

raylu commented:
OK, there's no reflection from `dir()` now, but I still don't see why `comic_scraper.py` needs to exist.
